### PR TITLE
PP-5133 Add pact test for zero amount not allowed error

### PR DESCRIPTION
**Pull request by stephencdaly**
Test the contract with connector that connector will give the expected error response when a request is made with a 0 amount and the flag on the gateway account is not enabled to allow this.

### CODE REVIEW

File: src/test/java/uk/gov/pay/api/service/CreatePaymentServiceTest.java [hunk 1]
```diff
@@ -216,4 +219,24 @@ public void testCreatePaymentWithPrefilledCardholderDetails() {
         assertThat(billingAddress.getPostcode(), is("AB1 CD2"));
         assertThat(billingAddress.getCountry(), is("GB"));
     }
+    
+    @Test
+    @PactVerification({"connector"})
+    @Pacts(pacts = {"publicapi-connector-create-payment-with-zero-amount-not-allowed"})
+    public void testCreatePaymentWithZeroAmountWhenZeroAmountNotAllowedForAccount() {
+        Account account = new Account("123456", TokenPaymentType.CARD);
+        ValidCreatePaymentRequest requestPayload = new ValidCreatePaymentRequest(CreatePaymentRequest.builder()
```
Comment (danworth) [inline src/test/java/uk/gov/pay/api/service/CreatePaymentServiceTest.java]: Just a heads up that @oswaldquek is likely to be removing `ValidCreatePaymentRequest` and just `CreatePaymentRequest` will remain but it'll be easy to update this test when that time comes.
